### PR TITLE
Add 'get_sdo' and 'get_sdo_entry' methods

### PR DESCRIPTION
**Pull request by flosse**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -186,6 +186,63 @@ pub struct SdoIndex {
...
+    pub op: Access,
+}
+
+impl From<(u8, u8)> for Access {
+    fn from((read, write): (u8, u8)) -> Self {
+        match (read, write) {
+            (0, 1) => Access::ReadOnly,
+            (1, 0) => Access::WriteOnly,
+            (1, 1) => Access::ReadWrite,
+            _ => Access::Unknown,
```
Comment (flosse) [inline src/types.rs]: @birkenfeld Do you know what it means if e.g. both are zero? Is it unknown or not accessible at all?

Comment (birkenfeld) [reply src/types.rs]: I don't know, but looking at the master code:

* the access mode is read directly from the device
* the mode doesn't seem to be checked by the master anywhere else

So a badly implemented device could presumably return (0,0) for the access but still process read and/or write requests when the respective ioctls are called.

File: src/master.rs [hunk 2]
```diff
@@ -336,6 +373,15 @@ impl Master {
     // reference_clock_time, sync_monitor_queue, sync_monitor_process
 }
 
+fn raw_array_to_string(arr: &[i8; 64]) -> String {
+    let s = arr
+        .iter()
+        .filter(|x| **x != 0)
```
Comment (birkenfeld) [inline src/master.rs]: why are NULs filtered, is it not a NUL terminated string?

File: src/master.rs [hunk 3]
```diff
@@ -336,6 +373,15 @@ impl Master {
...
+        .map(|x| *x as u8)
+        .collect::<Vec<_>>();
+    String::from_utf8_lossy(&s).to_string()
```
Comment (birkenfeld) [inline src/master.rs]: I'm not sure of the encoding here, although it's probably not UTF-8 :)  I suspect latin1 or windows1252.

File: src/master.rs [hunk 4]
```diff
@@ -273,6 +273,43 @@ impl Master {
...
+            max_subindex: sdo.max_subindex,
+            object_code: sdo.object_code,
+            name: raw_array_to_string(&sdo.name),
+        })
+    }
+
+    pub fn get_sdo_entry(
+        &mut self,
+        slave_index: u16,
+        sdo_spec: i32,
```
Comment (birkenfeld) [inline src/master.rs]: the format for `sdo_spec` seems to be nontrivial: if I read correctly, negative numbers are "by position", positive are sdo indices. That should be wrapped in an enum on the Rust side.

Maybe even include the subindex?

File: src/types.rs (hunk 1 above)
Comment (flosse) [reply src/types.rs]: ok, so would you agree with the `Unknown` variant here?

File: src/master.rs (hunk 2 above)
Comment (flosse) [reply src/master.rs]: You're probably right ;-) I'll have second look at it.

File: src/master.rs (hunk 3 above)
Comment (flosse) [reply src/master.rs]: good point. thx.

File: src/master.rs (hunk 4 above)
Comment (flosse) [reply src/master.rs]: :+1:

File: src/types.rs (hunk 1 above)
Comment (birkenfeld) [reply src/types.rs]: Yes!

File: src/master.rs (hunk 2 above)
Comment (flosse) [reply src/master.rs]: No I use your implementation and it seems to work :)
```rust
unsafe { CStr::from_ptr(data).to_string_lossy().into_owned() }
```

File: src/master.rs (hunk 4 above)
Comment (flosse) [reply src/master.rs]: I added the enum `AdoEntryAddr`. What do you think?

File: src/master.rs [hunk 5]
```diff
@@ -273,6 +269,52 @@ impl Master {
...
+            slave_position: sdo.slave_position,
+            position: sdo.sdo_position,
+            index: sdo.sdo_index,
+            max_subindex: sdo.max_subindex,
+            object_code: sdo.object_code,
+            name: c_array_to_string(sdo.name.as_ptr()),
+        })
+    }
+
+    pub fn get_sdo_entry(&mut self, slave_index: u16, addr: SdoEntryAddr) -> Result<SdoEntry> {
```
Comment (flosse) [inline src/master.rs]: what about shorten the fn to `sdo_entry`?

File: src/master.rs [hunk 6]
```diff
@@ -273,6 +269,52 @@ impl Master {
         })
     }
 
+    pub fn get_sdo(&mut self, slave_index: u16, sdo_position: u16) -> Result<SdoInfo> {
```
Comment (flosse) [inline src/master.rs]: what do you think about removing the `get` and just call it `sdo`?

**Pull request merged**